pkg/platform/kube/test/kubectlclient.go

```diff
@@ -0,0 +1,77 @@
+package test
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/nuclio/nuclio/pkg/cmdrunner"
+
+	"github.com/nuclio/logger"
+)
+
+type RunKubectlCommandMode string
+
+const (
+	RunKubectlCommandDirect   RunKubectlCommandMode = "direct"
+	RunKubectlCommandMinikube RunKubectlCommandMode = "minikube"
+)
+
+type RunOptions struct {
+	*cmdrunner.RunOptions
+	mode         RunKubectlCommandMode
+	modeExecutor string
+}
+
+func NewRunOptions(mode RunKubectlCommandMode, modeExecutor string) *RunOptions {
+	runOptions := &RunOptions{
+		mode:         mode,
+		modeExecutor: modeExecutor,
+	}
+	return runOptions
+}
+
+func RunKubectlCommand(logger logger.Logger,
+	cmdrunner cmdrunner.CmdRunner,
+	positionalArgs []string,
+	namedArgs map[string]string,
+	runOptions *RunOptions) (cmdrunner.RunResult, error) {
+
+	if runOptions == nil {
+		runOptions = NewRunOptions(RunKubectlCommandDirect, "kubectl")
+	}
+
+	var argsStringSlice []string
+
+	switch runOptions.mode {
+	case RunKubectlCommandDirect:
+		argsStringSlice = append(argsStringSlice, runOptions.modeExecutor)
+	case RunKubectlCommandMinikube:
+		argsStringSlice = append(argsStringSlice, runOptions.modeExecutor)
+	default:
+		argsStringSlice = append(argsStringSlice, runOptions.modeExecutor)
+	}
```

Comment:
You're doing the same for all cases.
Is that a preperation for something else?

Reply:
Fixed